feat: extend dryRun options to extract lookup full (#14555)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

## dryRun

> `dryRun` Preview the behavior of Renovate in logs, without making any changes to the repository files

Reviewer: ```suggestion
`dryRun` Preview the behavior of Renovate in logs, without making any changes to the repository files.
```

## dryRun

> "extract": Performs extract only, prints results, exits.<br>
> "lookup": Performs extract and lookup, but exits before running through branches.<br>
> "full": Perform a dry run by logging messages instead of creating/updating/deleting branches and PRs.<br>
> Default value = null.

Reviewer: ```suggestion
You can choose from the following behaviors for the `dryRun` config option:

- `null`: REPLACE WITH EXPLANATION, this is the default value
- `"extract"`: Performs extract only, prints results, exits
- `"lookup"`: Performs extract and lookup, but exits before running through branches
- `"full"`: Performs a dry run by logging messages instead of creating/updating/deleting branches and PRs.
```

Changes:

- Convert to list
  - Make `null` default the top list item
- Small rewrite so that the list items start with `performs`
- Add `monospaced` font to the config option strings
- Remove `<br>` HTML code

After applying my suggestion, please replace the `REPLACE WITH EXPLANATION` text with a proper explanation of what `null` means. 😉

## dryRun

> You can choose from the following behaviors for the `dryRun` config option:

Reviewer: Can you add a sentence above this one that explains what a "dry run" is, and why you might want to use it?

I guess it's a way to "preview" the behavior of Renovate, which doesn't actually make any changes to the repository files?

- `null`: Performs a regular renovate run include creating/updating/deleting branches and PRs, this is the default value

> - `"extract"`: Performs extract only, prints results, exits

Reviewer: I'm not sure what "extract only" means here, do we need to explain that in more detail here?

> - `"lookup"`: Performs extract and lookup, but exits before running through branches

Reviewer: Do we need to explain the lookup as well?

> - `"full"`: Performs a dry run by logging messages instead of creating/updating/deleting branches and PRs.

Reviewer: ```suggestion
- `"full"`: Performs a dry run by logging messages instead of creating/updating/deleting branches and PRs
```

We don't use punctuation at the end of list items.

## dryRun

> `dryRun` Preview the behavior of Renovate in logs, without making any changes to the repository files.

Reviewer: ```suggestion
Use `dryRun` to preview the behavior of Renovate in logs, without making any changes to the repository files.
```
